### DISCUSSION TITLE
组件编译到Web Component

### DISCUSSION THREAD

**Discussion by JerryChan31** [Tencent/tdesign#86] [Ideas]
![image](https://user-images.githubusercontent.com/14790315/149443028-c29ec3a4-1801-4c9f-9cc1-f903f301ea7f.png)

看到tdesign支持这么多端和框架，可以看出tdesign团队的宏大愿景和野心。

为什么实现方案上不考虑一下将某一个框架的组件编译到Web Component呢？看起来可以减少很多重复工作量，想了解一下贵团队是怎么考虑的。提前感谢指教！

**Comment by 94dreamer** [comment]
@xiaosansiji

**Comment by simpleAndElegant** [comment]
同问 为什么没考虑web components呢

**Comment by xiaosansiji** [comment]
提供这些框架支持是为了满足腾讯公司内部的实际使用需要，web components 在开发工具生态等方面跟 Vue/React 等还有比较大差距，线上项目完全使用 web components 开发的还不多。不过我们在 TDesign 官网尝试了下感觉还可以，现在官网的公共部分渲染已经是用 web components 开发了，也考虑后面增加下 TDesign 组件库的 web components 版本。
官网公共样式 web components 包：https://www.npmjs.com/package/tdesign-site-components

**Reply by hinagiku** [reply]
请问现在web components版本已经适配多少组件了？达到可用程度了吗
